web/src/todo_widget.ts

```diff
@@ -1,98 +1,88 @@
 import $ from "jquery";
+import assert from "minimalistic-assert";
+import {z} from "zod";
 
 import render_widgets_todo_widget from "../templates/widgets/todo_widget.hbs";
 import render_widgets_todo_widget_tasks from "../templates/widgets/todo_widget_tasks.hbs";
 
 import * as blueslip from "./blueslip";
 import {$t} from "./i18n";
+import type {Message} from "./message_store";
 import {page_params} from "./page_params";
 import * as people from "./people";
-import {todo_widget_extra_data_schema} from "./submessage";
+import type {Event} from "./poll_widget";
+
+export const todo_widget_extra_data_schema = z
+    .object({
+        task_list_title: z.string().optional(),
+        tasks: z.array(z.object({task: z.string(), desc: z.string()})).optional(),
+    })
+    .nullable();
+
+const todo_widget_inbound_data = z.intersection(
+    z.object({
+        type: z.enum(["new_task", "new_task_list_title", "strike"]),
+    }),
+    z.record(z.string(), z.unknown()),
+);
+
+const new_task_inbound_data_schema = z.object({
+    type: z.literal("new_task"),
+    key: z.number(),
```

Comment:
You can use `.int`, `.min` and `.max` for `key` here and remove this check later when parsing the data.
```ts
if (!Number.isInteger(idx) || idx < 0 || idx > MAX_IDX) {
       blueslip.warn("todo widget: bad type for inbound task idx");
       return;
 }
```

Reply:
Oops!! Yes we can. Fixed now!